File: pure-admin-thin/src/api/sysPayMethod.ts
import { http } from "@/utils/http";

export type SysPayMethod = {
  id: number;
  createdAt: string;
  updatedAt: string;
  methodCode: string;
  methodName: string;
  icon?: string | null;
  sort: number;
  status: number;
  remark?: string | null;
};

export type SysPayMethodSearch = {
  currentPage: number;
  pageSize: number;
  methodCode?: string;
  methodName?: string;
  status?: number | null;
};

export type SysPayMethodSet = {
  id?: number;
  methodCode: string;
  methodName: string;
  icon?: string | null;
  sort: number;
  status: number;
  remark?: string | null;
};

type SysPayMethodListResult = {
  code: number;
  message: string;
  success: boolean;
  data: {
    list: SysPayMethod[];
    total: number;
    pageSize: number;
    currentPage: number;
  };
};

type SysPayMethodResult = {
  code: number;
  message: string;
  success: boolean;
  data: SysPayMethod;
};

export const getSysPayMethodList = (data: SysPayMethodSearch) => {
  return http.request<SysPayMethodListResult>(
    "post",
    "/api/v1/admin/sysPayMethod/list",
    { data }
  );
};

export const getSysPayMethodById = (id: number) => {
  return http.request<SysPayMethodResult>(
    "get",
    `/api/v1/admin/sysPayMethod/${id}`
  );
};

export const setSysPayMethod = (data: SysPayMethodSet) => {
  return http.request<SysPayMethodResult>(
    "post",
    "/api/v1/admin/sysPayMethod",
    { data }
  );
};

export const delSysPayMethod = (id: number) => {
  return http.request<SysPayMethodResult>(
    "delete",
    `/api/v1/admin/sysPayMethod/${id}`
  );
};

File: core/repository/sys_pay_method.go
package repository

import (
	"BaseGoUni/core/pojo"
	"errors"
	"github.com/jinzhu/copier"
	"gorm.io/gorm"
	"time"
)

// GetSysPayMethods 支付方式列表（分页）
func GetSysPayMethods(db *gorm.DB, search pojo.SysPayMethodSearch) (result pojo.SysPayMethodResp) {
	var list []pojo.SysPayMethod
	query := db.Model(&pojo.SysPayMethod{}).Where("deleted_at = 0")

	if search.MethodCode != "" {
		query = query.Where("method_code LIKE ?", "%"+search.MethodCode+"%")
	}
	if search.MethodName != "" {
		query = query.Where("method_name LIKE ?", "%"+search.MethodName+"%")
	}
	if search.Status != nil {
		query = query.Where("status = ?", *search.Status)
	}

	query.Count(&result.Total)
	query = query.Order("sort asc, id desc").Limit(search.PageSize).Offset(search.PageSize * search.CurrentPage)
	query.Find(&list)

	for _, item := range list {
		var temp pojo.SysPayMethodBack
		_ = copier.Copy(&temp, &item)
		result.List = append(result.List, temp)
	}

	result.PageSize = search.PageSize
	result.CurrentPage = search.CurrentPage
	return result
}

// SetSysPayMethod 创建或更新支付方式
func SetSysPayMethod(db *gorm.DB, req pojo.SysPayMethodSet) (result pojo.SysPayMethodBack, err error) {
	var entity pojo.SysPayMethod
	if req.ID > 0 {
		db.Where("id = ? AND deleted_at = 0", req.ID).First(&entity)
		if entity.ID == 0 {
			return result, errors.New("更新的数据不存在")
		}
		_ = copier.Copy(&entity, &req)
		err = db.Save(&entity).Error
	} else {
		_ = copier.Copy(&entity, &req)
		err = db.Create(&entity).Error
	}
	if err != nil {
		return result, err
	}
	_ = copier.Copy(&result, &entity)
	return result, nil
}

// DelSysPayMethod 软删除支付方式
func DelSysPayMethod(db *gorm.DB, id int64) (result string, err error) {
	var entity pojo.SysPayMethod
	db.Where("id = ? AND deleted_at = 0", id).First(&entity)
	if entity.ID == 0 {
		return result, errors.New("删除的数据不存在")
	}
	err = db.Model(&entity).Update("deleted_at", time.Now().Unix()).Error
	if err != nil {
		return result, err
	}
	return "success", nil
}

// GetSysPayMethodById 根据ID获取支付方式
func GetSysPayMethodById(db *gorm.DB, id int64) (result pojo.SysPayMethodBack, err error) {
	var entity pojo.SysPayMethod
	db.Where("id = ? AND deleted_at = 0", id).First(&entity)
	if entity.ID == 0 {
		return result, errors.New("数据不存在")
	}
	_ = copier.Copy(&result, &entity)
	return result, nil
}
